### DISCUSSION TITLE
I want to change the pages directory address through the browser device

### DISCUSSION THREAD

**Discussion by carter911** [nuxt/nuxt#20044] [Questions]
I would like to ask a question. My current project has PC and mobile terminals. They are implemented separately. At present, I want to dynamically switch the pages directory according to the device type in app.vue. I have tried but it has not been implemented. No Any idea if there is a way around this
```
// app.vue
const template = ref("template/web");
useAppConfig({
  pages:template,
  dir:{
    pages:template
  }
})
```
The code above didn't work

**Comment by eggsy** [comment]
What you could do instead is wrap your views in separate folders:

```
├─ /views
│  ├─ mobile
│  	  └─ index.vue
└─ ├─ desktop
      └─ index.vue
```

and redirect requests via a router middleware or simply a page script that watches your mobile detection variable and replaces the current route with the target route:

```js
// app.vue
const router = useRouter();
const isCreated = ref(false);

onMounted(() => {
  isCreated.value = true;
});

const isMobile = computed(() => {
  if (!isCreated.value) return false;
  return /iPhone|iPad|iPod|Android/i.test(navigator.userAgent);
});

watch(
  () => isMobile.value,
  (isMobile) => {
    console.log(`{isMobile} value changed to: ${isMobile}`);
    if (isMobile) router.replace(`/mobile/${router.currentRoute.path}`);
  }
);
```

You could always use an external library but here I made it simply with Vue hooks. I am not sure if this is the best approach but this is the approach I'd go with.
